edgelet/edgelet-http/src/lib.rs

```diff
@@ -202,10 +208,10 @@ where
     <S::Service as Service>::Future: Send + 'static,
 {
     pub fn run(self) -> Run {
-        self.run_until(future::empty())
+        self.run_until(future::empty(), &ConcurrencyThrottling::NoLimit)
     }
 
-    pub fn run_until<F>(self, shutdown_signal: F) -> Run
+    pub fn run_until<F>(self, shutdown_signal: F, concurrency: &ConcurrencyThrottling) -> Run
```

Comment:
`concurrency: ConcurrencyThrottling`, no `&`, and put a `#[derive(Clone, Copy)]` on the enum.

Reply:
Ok. Out of curiosity, what is the difference?

Reply:
I though clippy told me to put & to avoid copying&cloning

Reply:
Fixed it too

Reply:
It told you to put the `&` because you didn't tell it the type is `Copy`, yes.

It's good to make small Copy-able types like this `Copy` because otherwise you're passing a reference to something that's easier to just pass by value.

Reply:
I think as a rule of thumb (and holds from C conventions), if it is immutable and a primitive (equal or less in size to a pointer) then passing by value or copy is fine. It is only when things are getting larger that passing by reference should be preferred (less stack space). But yes, Clippy will ask you to throw `&` at a lot of things that could probably be passed by value :)

Reply:
Thanks!